### PR TITLE
User & Github Profile Module

### LINKED ISSUE #243: [FEATURE] Create User-Entity
### ⭐ Suggestion

Create a new entity with the following properties

- Key: Discord-Username
- githubUsername
- bio
- type
- location

Location should be tracked with the `githubUsername` via the Github-API

### 💻 Use Cases

_No response_

### ❌ Related Problems

_No response_

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #243 

- [x] Add Token Authentication
- [x] Add e2e-Tests

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/api/pull/244"></a>

### CODE REVIEW

File: src/user/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      return await createdUser.save();
+    } catch (error) {
+      throw new HttpException(error.message, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  public async patch(userDTO: PatchUserDTO, id: string) {
+    let githubProfile = null;
+    if (userDTO.githubUsername) {
+      githubProfile = await this.githubProfileService.create(
```
Comment (Cahllagerfeld) [inline src/user/user.service.ts:42]: This needs to be adjusted. In our use-case we probably need to delete the embeded document from the collection and recreate it.

File: src/user/schema/user.schema.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  long: number;
+}
+
+@Schema({ _id: false, timestamps: false })
+export class GithubProfile extends Document {
+  @Prop({ required: true })
+  _id: string;
+
+  @Prop({ type: Location })
+  location: Location;
```
Comment (Cahllagerfeld) [inline src/user/schema/user.schema.ts:21]: In my opinion this needs to go to the client. There might be the scenario where someone updates this and then its not accurate here. We should store this redundantly. When making the API-Call the client needs to call the github-api to get the location.

**Review by Cahllagerfeld — COMMENTED** [review]
First small comments

**Comment by Cahllagerfeld** [comment]
Linting will be fixed not in this PR

**Comment by Cahllagerfeld** [comment]
I tested out some stuff, thats why I needed some reverts within this PR. It shouldn't be a problem when it gets squashed though.

**Comment by Cahllagerfeld** [comment]
I'll address the linting stuff in a seperate PR once, that PR is done

File: src/user/dto/patch-user.dto.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  @IsOptional()
+  @ApiProperty()
+  avatar: string;
+
+  @IsString()
+  @IsOptional()
+  @ApiProperty({ enum: types })
+  @IsIn(types)
+  type: string;
+}
```
Comment (eddiejaoude) [inline src/user/dto/patch-user.dto.ts:26]: Great idea for a patch version 👍

**Pull request merged**